### PR TITLE
Code cleanup in binder

### PR DESCRIPTION
**Pull request by vitek-karas**
I was reading through the binder and related code and fixed mostly formatting issues. There are some simple changes which remove unnecessary variables, stale comments and so on.

### CODE REVIEW

File: src/binder/assemblybinder.cpp [hunk 1]
```diff
@@ -1287,7 +1287,7 @@ namespace BINDER_SPACE
     }
     
     /*
-     * BindByTpaList is the entry-point for the custom binding algorithm on the Phone.
```
Comment (lpereira) [inline src/binder/assemblybinder.cpp]: ...on the Phone? Is this just someone writing the comment while receiving a call on the phone, or was this code somehow related to Windows Phone initially?

File: src/binder/assemblybinder.cpp [hunk 2]
```diff
@@ -1425,23 +1423,19 @@ namespace BINDER_SPACE
                             {
                                 fileName.Append(W(".ni.dll"));
                                 hr = GetAssembly(fileName,
-                                    fInspectionOnly,
-                                    FALSE, /* fIsInGAC */
-                                    TRUE /* fExplicitBindToNativeImage */,
-                                    &pAssembly);
+                                                 fInspectionOnly,
+                                                 FALSE, // fIsInGAC
```
Comment (lpereira) [inline src/binder/assemblybinder.cpp]: I don't know what's the usual guideline here, but, unless a boolean parameter to a function has very obvious meaning (e.g. `Widget::SetIsVisible(bool)`), it's a good idea to pass enum types as parameters, even if they have only two possible options.  It's a lot clearer from the call site what's happening, without requiring comments (you also get help from the compiler if you pass a value that doesn't apply to that particular enum, which is nice).

File: src/binder/assemblybinder.cpp (hunk 1 above)
Comment (vitek-karas) [reply src/binder/assemblybinder.cpp]: Funnily enough lot of the binder code we're using today comes from the Windows Phone code base   😄 .
I didn't touch this comment - there are many places where the comments are off in certain ways.. it's just that I don't feel comfortable yet to fix them up as I don't have full understanding of the area.

File: src/binder/assemblybinder.cpp (hunk 2 above)
Comment (vitek-karas) [reply src/binder/assemblybinder.cpp]: I agree to a point:
* declaring the enum adds non-trivial cost in terms of code lines and so on... and always makes me wonder if it has some hidden meaning
* it's just not how the code is written in the runtime - which doesn't mean we should not look at changing it, just that we would be introducing something relatively "new" and potentially confusing

Comment (jkotas) [reply src/binder/assemblybinder.cpp]: In this particular case - It is likely that most of these flags can be just deleted after further analysis, e.g. there is no GAC concept in CoreCLR, why do we have a flag for it?

Comment (vitek-karas) [reply src/binder/assemblybinder.cpp]: I've looked into that a little bit - we're using it as "IsTPAAssembly" basically - and there's a piece of logic that depends on it: https://github.com/dotnet/coreclr/blob/54d9d9b2698c9628f43494e46f855996b31838bd/src/binder/clrprivbindercoreclr.cpp#L160

Maybe there's a way to remove that as well... but for this change I just didn't want to go into all these details... yet.

**Pull request merged**